coderd/workspaceagents.go

```diff
@@ -447,7 +447,10 @@ func (api *API) workspaceAgentStartupLogs(rw http.ResponseWriter, r *http.Reques
 	api.WebsocketWaitGroup.Add(1)
 	api.WebsocketWaitMutex.Unlock()
 	defer api.WebsocketWaitGroup.Done()
-	conn, err := websocket.Accept(rw, r, nil)
+	conn, err := websocket.Accept(rw, r, &websocket.AcceptOptions{
+		// Vite in development mode can't cross-access this port.
+		OriginPatterns: []string{"localhost:8080"},
```

Comment:
I wonder if we should only conditionally allow this, e.g. when used in development setup. We could add a hidden server config option used by the develop.sh script?

@deansheather do you have any thoughts re: security?

Reply:
> We could add a hidden server config option used by the develop.sh script?

Do you know if we have any common `--dev` flag? I believe that this is one of the command options, you need to have always enabled for development purposes. Also, I don't know if we need more origins. I used to develop mostly on my mac, so I'm fine with localhost, but not sure about remote setups.

Reply:
Not as I'm aware. I was thinking that by setting `Hidden: true` in `codersdk/deployment.go` we could have it without being something officially supported. The idea isn't to make it configurable per-se, but it's a potentially nice side-effect.

Reply:
Ok 👍  I will prepare a new hidden flag, `--websocket-origin-pattern`.

Reply:
Can we not just have the vite dev server proxy the websockets to the backend and rewrite the origin/host?

Reply:
> Can we not just have the vite dev server proxy the websockets to the backend and rewrite the origin/host?

Vite maye expose handlers to overwrite these, but I haven't tried that approach yet. Should we do this conditionally based on `NODE_ENV == development`?

Reply:
Ok, I've applied the rewrite based on the Vite handler. For `NODE_ENV == development`, the Origin is rewritten to `CODER_HOST`. 

```
GET /api/v2/workspaceagents/3e880b8c-249d-45d6-81c8-8fd6fe7fda69/startup-logs?follow&after=0 HTTP/1.1
sec-websocket-extensions: permessage-deflate; client_max_window_bits
sec-websocket-key: vBQ4P1xnXr6tboGAY+eSTw==
cookie: csrf_token=Go2RELfFjglGukx+wZ4m23+uGXTwh8Vzky3wjn9xFAQ=; coder_session_token=Lm2xu8wKdE-0UBBBmj5BwbDeyuyBjqQ0k
accept-language: pl,en-US;q=0.9,en;q=0.8,pl-PL;q=0.7
accept-encoding: gzip, deflate, br
sec-websocket-version: 13
origin: http://127.0.0.1:3000
upgrade: websocket
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36
cache-control: no-cache
pragma: no-cache
connection: Upgrade
host: 127.0.0.1:3000
```